pkg/prometheus/promcfg.go

```diff
@@ -623,6 +630,41 @@ func (cg *ConfigGenerator) Generate(
 	return yaml.Marshal(cfg)
 }
 
+func (cg *ConfigGenerator) appendStorageSettingsConfig(cfg yaml.MapSlice, p *v1.Prometheus) (yaml.MapSlice, error) {
+	isExemplarsEnabled := false
+	for _, featureFlag := range p.Spec.EnableFeatures {
+		if featureFlag == "exemplar-storage" {
+			isExemplarsEnabled = true
+			break
+		}
+	}
+
+	if !isExemplarsEnabled {
+		return cfg, nil
+	}
+
+	storageSettings := yaml.MapSlice{
+		{
+			Key: "storage",
+			Value: yaml.MapSlice{
+				{
+					Key: "exemplars",
+					Value: yaml.MapSlice{
+						{
+							Key:   "max_exemplars",
+							Value: p.Spec.Exemplars.MaxSize,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	cfg = append(cfg, storageSettings...)
```

Comment:
We need to check for supported versions.
```suggestion
	cfg = cg.WithMinimumVersion("2.29.0").AppendMapItem(cfg, "storage", ...)
```